pallets/providers/src/utils.rs

```diff
@@ -830,24 +827,47 @@ impl<T: pallet::Config> MutateProvidersInterface for pallet::Pallet<T> {
         Ok(())
     }
 
-    // Bucket specific functions:
     fn add_bucket(
         msp_id: MainStorageProviderId<T>,
         user_id: T::AccountId,
         bucket_id: BucketId<T>,
-        bucket_root: MerklePatriciaRoot<T>,
+        private: bool,
+        collection_id: Option<T::BucketNftCollectionId>,
     ) -> DispatchResult {
         // TODO: Check that the bucket does not exist yet
```

Comment:
I would rather do this in a separate PR once we have used a double mapping from MSP -> BucketId and actually check the right thing. But adding this would require more changes. I would want to keep this PR as concise as possible.

The main purpose of this PR was to add NFT functionality which required buckets to at least be created and present in the system. We can create a PR dedicated to bucket management after.